external/vsphere-policy/api/v1alpha1/computepolicy_types.go

```diff
@@ -0,0 +1,108 @@
+// © Broadcom. All Rights Reserved.
+// The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ComputePolicySpec defines the desired state of ComputePolicy.
+type ComputePolicySpec struct {
+	// +optional
+
+	// Description specifies the desired description of the policy.
+	Description string `json:"description,omitempty"`
+
+	// +optional
+
+	// PolicyID specifies the ID of the underlying vSphere compute policy, if
+	// one is associated with this IaaS object.
+	PolicyID string `json:"policyID,omitempty"`
+
+	// +optional
+	// +kubebuilder:default=Mandatory
+
+	// EnforcementMode specifies how the policy is enforced.
+	//
+	// The valid modes include: Mandatory and Optional.
+	EnforcementMode PolicyEnforcementMode `json:"enforcementMode,omitempty"`
+
+	// +optional
+
+	// Match is used to match workloads to which this policy should be applied.
+	//
+	// A mandatory policy with this field unset is applied to all workloads in
+	// the namespace.
+	Match *MatchSpec `json:"match,omitempty"`
+
+	// +optional
+	// +listType=set
+
+	// Tags specifies the names of the TagPolicy objects in the same namespace
+	// that contain the information about the vSphere tags used to activate this
+	// policy.
+	Tags []string `json:"tags,omitempty"`
```

Comment:
I got confused when looking at the tests that has both tagPolicy.spec.tags vs computePolicy.spec.tags . Is it better to rename this to TagPolicies?

Reply:
Ok got it. Reviewed the policyEvaluation.status.policy.tags definition again